app/process/watcher/crypto-transfer/crypto-transfer.go

```diff
@@ -157,32 +158,42 @@ func (ctw CryptoTransferWatcher) beginWatching(q *queue.Queue) {
 func (ctw CryptoTransferWatcher) processTransaction(tx transaction.HederaTransaction, q *queue.Queue) {
 	ctw.logger.Infof("New Transaction with ID: [%s]", tx.TransactionID)
 
-	var amount int64
+	var amount string
 	for _, tr := range tx.Transfers {
 		if tr.Account == ctw.accountID.String() {
-			amount = tr.Amount
+			amount = fmt.Sprint(tr.Amount)
 			break
 		}
 	}
 
+	wholeMemoCheck := regexp.MustCompile("^0x([A-Fa-f0-9]){40}-[1-9][0-9]+-[1-9][0-9]+$")
+
 	decodedMemo, e := base64.StdEncoding.DecodeString(tx.MemoBase64)
-	if e != nil || len(decodedMemo) < 42 {
+	if e != nil || len(decodedMemo) < 42 || !wholeMemoCheck.MatchString(string(decodedMemo)) {
```

Comment:
the number `42` comes from the length of the ethereum address `0x` + 40chars. In our case, we have at least:
- eth address (42 chars)
- `2` dashes (`-`) (2 chars)
- At least 1 char between the dashes for the `fee` and `gasPrice` (2 chars)

This means that we could safely check for `len(decodedMemo) < 46` right?

Reply:
yes, I did not account for the additional information